catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -494,4 +566,29 @@ public void process(Exchange exchange) {
       ThreadContext.bind(SECURITY.getSystemSubject());
     }
   }
+
+  @VisibleForTesting
+  public static void setInputTransformerIds(InputTransformerIds transformerIds) {
+    inputTransformerIds = transformerIds;
+  }
+
+  @VisibleForTesting
+  public void setInputTransformers(List<ServiceReference<InputTransformer>> inputTransformers) {
+    this.inputTransformers = inputTransformers;
+  }
+
+  @VisibleForTesting
+  public void setConfigDeleted(boolean deleted) {
+    configDeleted.set(deleted);
+  }
+
+  @VisibleForTesting
+  void setTransformerCheckPeriodMillis(long transformerCheckPeriodMillis) {
+    this.transformerCheckPeriodMillis = transformerCheckPeriodMillis;
+  }
+
+  @VisibleForTesting
+  void setTransformerWaitTimeoutMillis(long transformerWaitTimeoutMillis) {
```

Comment:
✏️ I think constructor arguments are clearer than `@VisibleForTesting` methods. You could refactor `transformerCheckPeriodMillis` and `transformerWaitTimeoutMillis` to constructor arguments so that `transformerCheckPeriodMillis` and `transformerWaitTimeoutMillis` can be `final` and so that these `@VisibleForTesting` methods aren't needed. If not should `transformerCheckPeriodMillis` and `transformerWaitTimeoutMillis` be `volatile`, even though the setters are only used in the unit tests right now?

Reply:
Made them into ctor arguments and got rid of the first ctor.